src/main/java/org/kitteh/irc/client/library/feature/sts/STSPropertiesStorageManager.java

```diff
@@ -0,0 +1,206 @@
+    }
+
+    /**
+     * Checks if a policy has been stored for the hostname.
+     *
+     * @param hostname the hostname to check
+     * @return whether the entry exists in the store
+     */
+    @Override
+    public boolean hasEntry(String hostname) {
+        this.pruneEntries();
+        return this.properties.containsKey(hostname);
+    }
+
+    /**
+     * Cleans up the outdated entries from the properties instance.
+     */
+    private void pruneEntries() {
+        Set<String> stagedRemovals = new HashSet<>();
+
+        for (Object keyObj : this.properties.keySet()) {
+            String hostname = (String) keyObj;
+
+            String value = this.properties.getProperty(hostname);
+            String[] components = value.split("; ");
+            ZonedDateTime dt = ZonedDateTime.parse(components[0], DATE_TIME_FORMATTER);
+            if (dt.isBefore(ZonedDateTime.ofInstant(Instant.now(), ZoneOffset.UTC))) {
+                stagedRemovals.add(hostname); // CME
+            }
+        }
+
+        stagedRemovals.forEach(this.properties::remove);
+        this.saveData();
+    }
+
+    /**
+     * Deletes an entry from the store (used for 0 duration policies).
+     * <p>
+     * Implementers MUST ignore requests to remove entries that do not exist.
+     *
+     * @param hostname the hostname to remove the policy for
+     */
+    @Override
+    public void removeEntry(String hostname) {
+        if (!this.hasEntry(hostname)) {
+            return;
+        }
+        this.properties.remove(hostname);
+    }
+
+    /**
+     * Reserialize the STS data in the same form as used in the spec.
+     *
+     * @param data The map of keys -> optional string values
+     * @return a serialized string
+     */
+    private String reserializeData(Map<String, Optional<String>> data) {
+        StringBuilder sb = new StringBuilder(data.keySet().size()*5);
+        Iterator<String> iterator = data.keySet().iterator();
+        while (iterator.hasNext()) {
+            String s = iterator.next();
+
+            sb.append(s);
+            final Optional<String> value = data.get(s);
+            if (value.isPresent()) {
+                sb.append("=").append(value.get());
+            }
+            if (iterator.hasNext()) {
+                sb.append(",");
+            }
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Returns an ISO8601 DateTime string using UTC.
+     *
+     * @param duration the time in seconds from now
+     * @return the date-time string e.g. 2016-01-01T00:00:00Z
+     */
+    private String getExpiryFromDuration(long duration) {
+        Instant now = Instant.now();
+        Instant then = now.plusSeconds(duration);
+        return ZonedDateTime.ofInstant(then, ZoneOffset.UTC).format(DATE_TIME_FORMATTER);
+    }
+
+
+    @Override
+    public void close() throws Exception {
+        System.out.println("Called close");
+        this.writer.flush();
```

Comment:
I'll change the code to use this, but it doesn't fix the test failure unfortunately. Likely I'll need to ditch `Writer`.